src/mpid/common/shm/mpidu_shm.h

```diff
@@ -0,0 +1,54 @@
+/* -*- Mode: C; c-basic-offset:4 ; indent-tabs-mode:nil ; -*- */
+/* vim: set ft=c.mpich : */
+/*
+ *  (C) 2019 by Argonne National Laboratory.
+ *      See COPYRIGHT in top-level directory.
+ */
+
+#ifndef MPIDU_SHM_H_INCLUDED
+#define MPIDU_SHM_H_INCLUDED
+
+/* This is a collective function: every process in the communicator comm must
+ * call it. The function expects the amount of memory to be allocated per node.
+ *
+ * If the node contains multiple processes the function tries to allocate symmetric
+ * shared memory (i.e., base address is the same on all processes).  If the node
+ * contains only one process the function allocates private memory for the process
+ * but still attempts to get the base address equal across all processes. If none
+ * of the processes in the node requests any memory they behave as if symmetric
+ * memory was allocated successfully.
+ *
+ * The function returns MPI_SUCCESS and the base address of the allocated memory
+ * if successful,  MPIDU_ERR_SYM_SHMEM_FAIL if symmetric shared memory could not
+ * be allocated or MPI_ERR_OTHER if any other error occurred.
+ *
+ * The off argument is used to shift the base address returned by the function so
+ * that the window base address for all processes in different nodes is the same.
+ * This optimization allows processes to perform data transfers without querying
+ * the target for the address of the buffer.
+ *
+ * It is up to the caller to map the process to the right position in shared memory
+ * or try another allocation mechanism if an error code is returned. */
+int MPIDU_shm_alloc_symm_all(MPIR_Comm * comm_ptr, size_t len, size_t off, void **ptr);
+
```

Comment:
Right but right now this is CH4 specific. It may still make sense to make it "sharable" among devices in case new ones get added in the future.

Reply:
Then the function and the comment need cleanup. I see several places use "win".

Reply:
Maybe, since it is only used by RMA currently, I should rename this function `MPIDU_shm_alloc_symm_win()`. We can change it later if we find another use case for it (?)

Reply:
It seems weird to put `MPIDU_shm_alloc_symm_win` in common. It is not a common routine anymore.

Reply:
ok, I have eliminated all references to win and rma